File: JS Front-End/02-JS Arrays and Strings/Exercise/qu1.js
function rotateArray(myArray, num) {
    for (let i = 1; i <= num; i++) {
        myArray.push(myArray.shift());
    }

    console.log(myArray.join(' '));
}

// rotateArray([51, 47, 32, 61, 21], 2);
// rotateArray([32, 21, 61, 1], 4);
// rotateArray([2, 4, 15, 31], 5);

File: Softuniada/2003/qu1.py
# Last row in Pascal's triangle

num = int(input())

current = []
initial_steps = [[1], [1, 1], [1, 2, 1]]

if num < 3:
    current = initial_steps[num]
else:
    initial = initial_steps[-1]

    for line in range(3, num + 1):
        current = [1]
        for i in range(1, line):
            middle = [initial[i] + initial[i - 1]]
            current.extend(middle)
        current.extend([1])
        initial = current

print(' '.join([str(num) for num in current]))
